libs/api/metadata-converter/src/lib/xml-utils.ts

```diff
@@ -225,14 +239,40 @@ export function xmlToString(
   const parentPadding = '    '.repeat(Math.max(0, indentationLevel - 1))
   if (children === '') {
     return `
-${padding}<${el.name}${attrs}/>
+${padding}<${el.name}${attrs}${gmlNamespace}/>
 ${parentPadding}`
   }
+
   return `
-${padding}<${el.name}${attrs}>${children}</${el.name}>
+${padding}<${el.name}${attrs}${gmlNamespace}>${children}</${el.name}>
 ${parentPadding}`
 }
 
+function getGmlNamespaceAsNeeded(
```

Comment:
I don't think you need that; the namespaces are added at the end when producing the XML document, see:
https://github.com/geonetwork/geonetwork-ui/blob/f92455971a26a1110d02acb31a142cc27e7cd6d8/libs/api/metadata-converter/src/lib/xml-utils.ts#L34

Reply:
Actually, the way I parse/serialize the GML part with OpenLayers, I don't have the full access to the XmlDocument at that time, as OpenLayers only takes in native node objects, I have to use a string as a intermediate step at some point.
It's there that I'm missing the GML namespace.
If you know a way to have OL understand the Xml elements from the lib we use to parse the record, I'd be really interested.
I agree adding this namespace manually is ugly (and the serialization is ugly too, we change default namespace in the middle of the tree, but apparently it's allowed, so...).

Reply:
I've tried a few things, I think this can be handled inside the `readGeometry` function without any change to `xmlToString`, see my other comment